specs/hotel_spec.rb

```diff
@@ -0,0 +1,242 @@
+require_relative 'spec_helper'
+# require 'pry'
+describe 'Hotel' do
+  before do
+    @hotel = Hotel::Hotel.new
+    @date1 = Date.new(2018,9,12)
+    @date2 = Date.new(2018,9,15)
+    @date3 = Date.new(2018,9,13)
+    @date4 = Date.new(2018,9,19)
+  end
+
+  describe 'initialize' do
+    it 'creates an instance of Hotel' do
+      @hotel.must_be_instance_of Hotel::Hotel
+    end
+
+    it 'creates a hash of rooms' do
+      @hotel.rooms.must_be_kind_of Array
+    end
+
+    it 'creates an array of reservations' do
+      @hotel.reservations.must_be_kind_of Array
+    end
+
+  end
+
+  describe 'make_reservation' do
+    it 'creates a reservation object with two dates and pushes to reservations' do
+      @hotel.make_reservation(date1: @date1, date2: @date2).must_be_kind_of Array
+    end
+    it 'creates a reservation object with one date and pushes to reservations' do
+      @hotel.make_reservation(date1: @date1).must_be_kind_of Array
+    end
+    it 'returns an array of reservation objects' do
+      @hotel.reservations.each do |reservation|
+        reservation.must_be_instance_of Hotel::Reservation
+      end
+    end
+end
+  describe 'view available' do
+    it 'returns a list of rooms that are available for a given date' do
+
+      @hotel.make_reservation(date1: @date1 , date2: @date2)
+      @hotel.view_available(date1: @date1).must_be_kind_of Array
+      @hotel.view_available(date1: @date1).must_equal [:room2, :room3, :room4, :room5, :room6, :room7, :room8, :room9, :room10,
+        :room11, :room12, :room13, :room14, :room15, :room16, :room17, :room18, :room19, :room20]
+      @hotel.view_available(date1: @date1, date2: @date3).must_equal [:room2, :room3, :room4, :room5, :room6, :room7, :room8, :room9, :room10,
+        :room11, :room12, :room13, :room14, :room15, :room16, :room17, :room18, :room19, :room20]
+      @hotel.view_available(date1: @date3, date2: @date4).must_equal [:room2, :room3, :room4, :room5, :room6, :room7, :room8, :room9, :room10,
+        :room11, :room12, :room13, :room14, :room15, :room16, :room17, :room18, :room19, :room20]
+      @hotel.make_reservation(date1: @date1, date2: @date2)
+      @hotel.view_available(date1: @date1, date2: @date2).must_equal [:room3, :room4, :room5, :room6, :room7, :room8, :room9, :room10,
+        :room11, :room12, :room13, :room14, :room15, :room16, :room17, :room18, :room19, :room20]
+    end
```

Comment:
Other cases I'd be interested in:
- What if there are no reservations? (it should return a list of all the rooms)
- Do reservations that don't overlap the date range affect the list? (they shouldn't)
- What if there are no vacancies? (it should return an empty array)